staging/src/k8s.io/client-go/kubernetes/register.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"os"
+
+	"k8s.io/apimachinery/pkg/apimachinery/announced"
+	"k8s.io/apimachinery/pkg/apimachinery/registered"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/pkg/apis/batch/install"
+)
+
+var groupFactoryRegistry = make(announced.APIGroupFactoryRegistry)
+var registry = registered.NewOrDie(os.Getenv("KUBE_API_VERSIONS"))
+var scheme = runtime.NewScheme()
```

Comment:
@deads2k your comment from irc was that `runtime.RawExtension` needs all types that should be embedded. E.g. in the PodTemplate case this is the pod.

We could make `scheme` public of course, or add some way to compose apigroups and clients explicitly.